pylint/checkers/base/basic_checker.py

```diff
@@ -453,7 +453,16 @@ def visit_expr(self, node: nodes.Expr) -> None:
 
         # Warn W0133 for exceptions that are used as statements
         if isinstance(expr, nodes.Call):
-            inferred = utils.safe_infer(expr)
+            name = ""
+            if isinstance(expr.func, nodes.Name):
+                name = expr.func.name
+            elif isinstance(expr.func, nodes.Attribute):
+                name = expr.func.attrname
+
+            # Heuristic: only run inference for names that begin with an uppercase char
+            # This reduces W0133's coverage, but retains acceptable runtime performance
+            # For more details, see: https://github.com/PyCQA/pylint/issues/8073
+            inferred = utils.safe_infer(expr) if name[:1].isupper() else None
```

Comment:
This is pretty clever in term of filtering, I would not have bet on this being good enough to improve performance, but I guess it makes sense as it remove all the function call when pep8 is followed.